### ISSUE TITLE
Rendering: changing the plottable list mid-render will cause an exception

### ISSUE THREAD

**Issue by ZSYMAX** [ScottPlot/ScottPlot#3753]
**Issue:** (It will cause collection has changed exception on Rendering)

**ScottPlot Version:** (5.0.31)

https://github.com/ScottPlot/ScottPlot/blob/45e90a9eacce88cbd8d58526b883fbac32a83f11/src/ScottPlot5/ScottPlot5/Rendering/RenderActions/RenderPlottables.cs#L7-L26

```cs
/* code to reproduce the bug */
 MeasuredSpan = avaplot.Plot.Add.HorizontalSpan(x1, x1, Colors.Red.WithOpacity(0.4));
 MeasuredSpan.Axes.XAxis = avaplot.Plot.Axes.Top;
```

when Plot.Add.HorizontalSpan() called,it will add to the PlottableList,if now is rendering,it will caused collection has changed exception. 
So rp.Plot.PlottableList need a  sample clone, for example rp.Plot.PlottableList.ToList()

**Comment by swharden** [comment]
Thanks for reporting this @ZSYMAX! Re-stating the issue, adding a new object to the plot while a plot is currently rendering will cause an exception. This is similar to what happens if data inside plottables is modified while a render is occurring, and some strategies for avoiding this issue are described here https://scottplot.net/faq/async/

However, your suggestion of copying the list at the start of a render is a good one! I will make this fix now 👍

**Comment by swharden** [comment]
Thanks again for the report @ZSYMAX! I think I fixed it in 3609298 which will be in the next release (ScottPlot 5.0.32) but if it doesn't work as expected let me know and we can open this up and take a closer look 👍
